src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -367,27 +370,46 @@ class OrderedRVD private(
     spec.write(sparkContext.hadoopConfiguration, path)
     partitionCounts
   }
-}
 
-object OrderedRVD {
-  type CoercionMethod = Int
+  def zipPartitions(that: RVD)
+    (zipper: (Iterator[RegionValue], Iterator[RegionValue]) => Iterator[RegionValue])
+      : OrderedRVD =
+    zipPartitions(that, true)(zipper)
+
+  def zipPartitions(that: RVD, preservesPartitioning: Boolean)
+    (zipper: (Iterator[RegionValue], Iterator[RegionValue]) => Iterator[RegionValue])
+      : OrderedRVD =
+    zipPartitions(that.rdd, preservesPartitioning)(zipper)
+
+  def zipPartitions[T: ClassTag](that: RDD[T])
+    (zipper: (Iterator[RegionValue], Iterator[T]) => Iterator[RegionValue])
+      : OrderedRVD =
+    zipPartitions(that, true)(zipper)
+
+  def zipPartitions[T: ClassTag](that: RDD[T], preservesPartitioning: Boolean)
+    (zipper: (Iterator[RegionValue], Iterator[T]) => Iterator[RegionValue])
+      : OrderedRVD =
+    OrderedRVD(
+      typ,
+      partitioner,
+      this.rdd.zipPartitions(that, preservesPartitioning)(zipper))
```

Comment:
It seems like the type should be given by `this.rowTyp ++ that.rowTyp`, not provided, right? If there exist conflicting fields the user can use `updateType` to resolve that.

Reply:
Your zipPartitions functions take a f: (Iterator[RV], Iterator[RV]) => Iterator[RV].  The type won't be the concat, but will depend on f.  If you want to concat instead of taking f, then this.rowTyp ++ that.rowTyp seems reasonable.

Reply:
I don't see how not taking a `zipper: (Iterator[RV], Iterator[RV]) => Iterator[RV]` is an option. Then you have to choose how to combine iterators, e.g. some kind of join, and then it's no longer a `zipPartitions` method. But then, as Cotton says, they need to take a type for the region values produced by `zipper`.

I used a similar pattern in #3159 to make `OrderedRVD.join` return an `OrderedRVD`. And when this goes in, I should refactor the join methods to use your `OrderedRVD.zipPartitions`.